src/core/core.interaction.js

```diff
@@ -293,7 +294,33 @@ module.exports = {
 			if (options.intersect && !intersectsItem) {
 				items = [];
 			}
-			return items;
+
+			// group by dataset
+			var byIndex = items.reduce(function(res, element) {
+				if (res[element._datasetIndex] === undefined) {
+					res[element._datasetIndex] = [element];
+				} else {
+					res[element._datasetIndex].push(element);
+				}
+				return res;
+			}, {});
+
+			// sort by distance
+			Object.keys(byIndex).forEach(function(key) {
+				byIndex[key].sort(function(a, b) {
+					var centerA = a.getCenterPoint();
+					var centerB = b.getCenterPoint();
+					var distanceA = distanceMetric(position, centerA);
+					var distanceB = distanceMetric(position, centerB);
+					return distanceA - distanceB;
```

Comment:
You could write these 5 lines as 1 line:

    return distanceMetric(position, a.getCenterPoint()) - distanceMetric(position, b.getCenterPoint());

That would result in smaller code for the user to download to the client